extension/src/background/index.ts

```diff
@@ -28,20 +28,20 @@ export const initContentScriptMessageListener = () => {
 };
 
 export const initExtensionMessageListener = (sessionStore: Store) => {
-  browser?.runtime?.onMessage?.addListener((request, sender) => {
+  browser?.runtime?.onMessage?.addListener(async (request, sender) => {
     // todo this is kinda ugly
     let res;
     if (Object.values(SERVICE_TYPES).includes(request.type as SERVICE_TYPES)) {
       // eslint-disable-next-line
-      res = popupMessageListener(request, sessionStore);
+      res = await popupMessageListener(request, sessionStore);
     }
     if (
       Object.values(EXTERNAL_SERVICE_TYPES).includes(
         request.type as EXTERNAL_SERVICE_TYPES,
       )
     ) {
       // eslint-disable-next-line
-      res = freighterApiMessageListener(request, sender, sessionStore);
+      res = await freighterApiMessageListener(request, sender, sessionStore);
```

Comment:
without an async listener, this wasn't waiting for promises within the popup and freighter-api message listener to resolve